testutil/kubernetes_helper.go

```diff
@@ -211,33 +211,31 @@ func (h *KubernetesHelper) CheckPods(ctx context.Context, namespace string, depl
 
 	err := h.retryFor(6*time.Minute, func() error {
 		checkedPods = []corev1.Pod{}
-		pods, err := h.clientset.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{})
+		pods, err := h.GetPodsForDeployment(ctx, namespace, deploymentName)
 		if err != nil {
 			return err
 		}
 
 		var deploymentReplicas int
-		for _, pod := range pods.Items {
-			if strings.HasPrefix(pod.Name, deploymentName) {
```

Comment:
Associating pods with their deployments by comparing prefixes was causing problems because the `jaeger` and `jaeger-collector` deployments share a prefix.  I changed this to reuse the GetPodsForDeployments method which relies on the deployment's label selector instead.